### PR TITLE
Minor ANSI color support

### PR DESCRIPTION
**Pull request by xxx**
I don't know if this feature is desired in the main trunk or not, but here it is.

We've been wanting to use color for awhile, (like 20-something years, heh) but `break_string` and the WORD_WRAP section within `write_socket` take the sequence characters into account when doing various length checks, so things are formatted poorly. This changeset just adds a define, `ANSI_COLOR`, which changes the behavior of the above 2 functions, along with `strlen`, to not take the sequence characters into account when checking lengths.

It's tightly focused on ANSI SGR sequences (so color and text formatting only). It assumes all further support of these codes is handled by the mudlib.

In #7, you had mentioned about read_file and write_bytes, but I left them as-is, since when files are concerned I think they should reflect everything read or written from them. strlen arguably doesn't need to be changed, either.

I haven't really written any C in a very long time. I'm probably ignorant about some things here. I tried to follow existing style.

### CODE REVIEW

File: config.h [hunk 1]
```diff
@@ -363,5 +363,16 @@
...
+/*
+ * ANSI_COLOR
+ *
+ * When defined, ANSI color sequences are ignored when
+ * calculating the length of strings for display
+ * (i.e. strlen & break_string efuns, and write_socket()).
+ * This functionality only handles ANSI SGR sequences.
+ * Further color support is to be handled by the mudlib.
+ */
+#undef ANSI_COLOR
```
Comment (xxx) [inline config.h:364]: off by default

File: interpret.h [hunk 2]
```diff
@@ -159,6 +159,11 @@ extern struct control_stack *csp;	/* Points to last element pushed */
 #define INCREF(x) if (x) x++
 #define DECREF(x) if (x) x--
 
+#ifdef ANSI_COLOR
```
Comment (xxx) [inline interpret.h:162]: Is there a better place for these?

File: regress/expected.output [hunk 3]
```diff
@@ -10,21 +10,21 @@ Loading /tests/test-008.c -> Ok.
 Loading /tests/test-009.c -> Ok.
 Loading /tests/test-010.c -> Ok.
 Loading /tests/test-011.c -> *too many alarms in object
-Loading /tests/test-012.c -> tests/test-012.c: redefinition of variable 'i' line 7
-tests/test-012.c: Type mismatch ( int * vs int ) when initializing i line 7
+Loading /tests/test-012.c -> /tests/test-012.c: redefinition of variable 'i' line 7
+/tests/test-012.c: Type mismatch ( int * vs int ) when initializing i line 7
```
Comment (xxx) [inline regress/expected.output:14]: A number of differences here are just the leading slash. I'm just running `make check` locally to run these, so not sure.

File: regress/expected.output [hunk 4]
```diff
@@ -34,9 +34,21 @@ Loading /tests/test-024.c -> Ok.
 Loading /tests/test-025.c -> Ok.
 Loading /tests/test-026.c -> Ok.
 Loading /tests/test-027.c -> *Too deep recursion
-Loading /tests/test-028.c -> Ok.
```
Comment (xxx) [inline regress/expected.output:37]: This implies there's a test-028, but it was never checked into source control. I left a spot for it and used -029 instead.

File: regress/tests/test-029.c [hunk 5]
```diff
@@ -0,0 +1,101 @@
+#pragma strict_types
```
Comment (xxx) [inline regress/tests/test-029.c:1]: I was able to test the strlen and break_string changes easily enough, but I have no idea how to test the change to write_socket.

File: comm1.c [hunk 6]
```diff
@@ -294,13 +303,43 @@ write_socket(char *cp, struct object *ob)
...
-	    if (current_column + len >= ip->screen_width)
+                    if (cp[len] == ANSI_END)
+                        in_ansi = 0;
+                }
+#endif
+                len++;
+            }
+
+#ifdef ANSI_COLOR
+            if (current_column + len - ansi_len >= ip->screen_width)
```
Comment (xxx) [inline comm1.c:333]: This is pretty much the pattern. Track how long the sequences are, and subtract that amount when doing length checks.

**Comment by xxx** [comment]
Erm, going to close this out. I'm running into cases where I want both versions of strlen available, and sprintf has not been taken into account at all, but probably needs to be.

**Comment by mindware** [comment]
@xxx I was able to add ansi color, like you I ran into the sprintf challenge. Let me know if you managed to address it.

**Pull request closed without merging**